stwcs/updatewcs/astrometry_utils.py

```diff
@@ -640,8 +658,18 @@ def find_gsc_offset(obsname, refframe="ICRS"):
         gsss_serviceLocation = gsss_url
 
     # Initialize variables for cases where no offsets are available.
-    delta_ra = delta_dec = None
-    delta_roll = delta_scale = None
+    delta_ra = delta_dec = 0.0
+    delta_roll = 0.0
+    delta_scale = 1.0
+    dGSinputRA = dGSoutputRA = 0.0
+    dGSinputDEC = dGSoutputDEC = 0.0
+    outputCatalog = None
+
+    # Insure input is a fits.HDUList object, if originally provided as a filename(str)
+    close_obj = False
+    if isinstance(obsname, str):
+        obsname = fits.open(obsname)
+        close_obj = True
```

Comment:
```suggestion
    if isinstance(obsname, str):
        obsname = fits.open(obsname)
        close_obj = True
    else:
        close_obj = False
```

Reply:
When would either implementation differ in the results?  The original implementation insures that `close_obj` always gets defined regardless of the `if` statement and is actually more compact.  Just curious...

Reply:
It is not about compactness but rather readability _and_ (in principle) performance: There is no need to always assign a value to a variable just to modify it almost immediately in the next `if`. `else` is created just for that.

Reply:
This variable always needs to be defined in order for the later logic to work as it uses this variable.  This, I thought, was actually more apparent in my original implementation.